src/stress/memory/oom.spec.ts

```diff
@@ -3,32 +3,157 @@ Stress tests covering robustness when available VRAM is exhausted.
 `;
 
 import { makeTestGroup } from '../../common/framework/test_group.js';
+import { unreachable } from '../../common/util/util.js';
+import { GPUConst } from '../../webgpu/constants.js';
 import { GPUTest } from '../../webgpu/gpu_test.js';
 import { exhaustVramUntilUnder64MB } from '../../webgpu/util/memory.js';
 
 export const g = makeTestGroup(GPUTest);
 
+function createBuffer(
+  device: GPUDevice,
+  size: number,
+  mappable: boolean,
+  mode: GPUMapModeFlags,
+  mappedAtCreation: boolean
+) {
+  if (!mappable && !mappedAtCreation) {
+    return device.createBuffer({
+      size,
+      usage: GPUBufferUsage.UNIFORM,
+      mappedAtCreation,
+    });
+  }
+  let buffer: GPUBuffer;
```

Comment:
so here, we could have `mappable = false` but `mappedAtCreation = true` - but then we're using MAP_READ or MAP_WRITE usage. that doesn't make sense

Reply:
This is intentional. Basically the idea is that the buffer is re-mappable, but not mappable in it's current state. For the case you mention, it is not mappable because it is already mapped.

Reply:
Ah, I see. Maybe we should call the function `createBufferWithMapState` and instead use the [states](https://gpuweb.github.io/gpuweb/#enumdef-gpubuffermapstate) defined in the spec.

Saying "create buffer with mappable = true` reads like you're giving it a mappable usage, not that it's in a mappable state.

Reply:
Done.